### PR TITLE
Refactor kubelet to code

### PR DESCRIPTION
**Pull request by justinsb**
More splitting up the CoreOS PR

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1523)

### CODE REVIEW

File: nodeup/pkg/model/kubelet.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+func (b *KubeletBuilder) Build(c *fi.ModelBuilderContext) error {
+	kubeletConfig, err := b.buildKubeletConfig()
+	if err != nil {
+		return fmt.Errorf("error building kubelet config: %v", err)
+	}
+
+	// Add sysconfig file
+	{
+		// TODO: Dump this - just complexity!
+		flags, err := flagbuilder.BuildFlags(kubeletConfig)
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubelet.go]: Do we have an issue?

File: nodeup/pkg/model/kubelet.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+		c.AddTask(t)
+	}
+
+	c.AddTask(b.buildSystemdService())
+
+	return nil
+}
+
+func (b *KubeletBuilder) kubeletPath() string {
+	kubeletCommand := "/usr/local/bin/kubelet"
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubelet.go]: Same questions about `const`, and appending the path.

File: nodeup/pkg/model/kubelet.go [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+		return "", fmt.Errorf("error marshalling kubeconfig to yaml: %v", err)
+	}
+
+	return string(yaml), nil
+}
+
+func (b *KubeletBuilder) buildKubeletConfig() (*kops.KubeletConfigSpec, error) {
+	instanceGroup := b.InstanceGroup
+	if instanceGroup == nil {
+		// Old clusters might not have exported instance groups
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubelet.go]: How old?  How does that impact upgrades?

File: nodeup/pkg/model/kubelet.go [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+			instanceGroup.Spec.Role = kops.InstanceGroupRoleNode
+		}
+		//b.InstanceGroup = instanceGroup
+	}
+	kubeletConfigSpec, err := kops.BuildKubeletConfigSpec(b.Cluster, instanceGroup)
+	if err != nil {
+		return nil, fmt.Errorf("error building kubelet config: %v", err)
+	}
+	// TODO: Memoize if we reuse this
+	return kubeletConfigSpec, nil
```
Comment (chrislovecnm) [inline nodeup/pkg/model/kubelet.go]: Should I ask again if we have an issue ;)

File: nodeup/pkg/model/kubelet.go (hunk 1 above)
Comment (justinsb) [inline nodeup/pkg/model/kubelet.go]: No, a TODO is appropriate here

File: nodeup/pkg/model/kubelet.go (hunk 2 above)
Comment (justinsb) [inline nodeup/pkg/model/kubelet.go]: Let's keep it simple

File: nodeup/pkg/model/kubelet.go (hunk 3 above)
Comment (justinsb) [inline nodeup/pkg/model/kubelet.go]: I don't know, but if we see it reported, we'll deal with it

File: nodeup/pkg/model/kubelet.go (hunk 4 above)
Comment (justinsb) [inline nodeup/pkg/model/kubelet.go]: We have a TODO

**Pull request merged**